lib/checkstl.cpp

```diff
@@ -267,15 +362,61 @@ void CheckStl::iterators()
                 invalidationScope = nullptr;
             }
 
-            // Is iterator compared against different container?
-            if (tok2->isComparisonOp() && containerToken && tok2->astOperand1() && tok2->astOperand2()) {
-                const Token *other = nullptr;
-                if (tok2->astOperand1()->varId() == iteratorId)
-                    other = tok2->astOperand2()->tokAt(-3);
-                else if (tok2->astOperand2()->varId() == iteratorId)
-                    other = tok2->astOperand1()->tokAt(-3);
-                if (Token::Match(other, "%name% . end|rend|cend|crend ( )") && other->varId() != containerToken->varId())
-                    iteratorsError(tok2, getContainerName(containerToken), getContainerName(other));
+            // Is iterator compared against different container or iterator of different container?
```

Comment:
I meant to put all this code below

    // Is iterator compared against different container or iterator of different container?

in a function.

So the code here will just be something like:

    // Is iterator compared against different container or iterator of different container?
    if (foundOperatorErrors.find(tok2) == foundOperatorErrors.end() && compareIteratorAgainstDifferentContainer(tok2, containerToken))
        foundOperatorErrors.insert(tok2);

Reply:
@danmar I have put all code in a function. Looks better, but I'm thinking to make `iteratorScopeBeginInfo` as a class field (for not passing it in new method `compareIteratorAgainstDifferentContainer`), on the other hand, it's used only for iterator checker. Maybe in future to do it, when more methods will be needed this info.